mr/src/main/java/org/elasticsearch/hadoop/util/StringUtils.java

```diff
@@ -126,20 +127,44 @@ public static List<String> tokenize(String string, String delimiters, boolean tr
         if (!StringUtils.hasText(string)) {
             return Collections.emptyList();
         }
-        StringTokenizer st = new StringTokenizer(string, delimiters);
-        List<String> tokens = new ArrayList<String>();
-        while (st.hasMoreTokens()) {
-            String token = st.nextToken();
-            if (trimTokens) {
-                token = token.trim();
+        List<String> tokens = new ArrayList<>();
+        char[] delims = delimiters.toCharArray();
+        StringBuilder currentToken = new StringBuilder();
+        boolean inQuotedToken = false;
+        for (char character : string.toCharArray()) {
+            if (character == '\"') {
+                inQuotedToken = !inQuotedToken;
```

Comment:
I still think just going in an encoding/decoding the fields where needed would be simpler. For instance, this logic now breaks if we include quotation marks in the field names. Granted, that's probably even more unlikely to happen than commas, but if we're talking about respecting the tokenize method contracts, it's still broken.

Reply:
I'll have to come back to this when I have a little time to get into it. I don't remember exactly what I meant by
```
it does something that I don't think most people would expect of methods called tokenize and concatenate
```
But I remember it being bad enough that it seemed like a deal-killer for encoding/decoding the fields.